src/basic/terminal-util.c

```diff
@@ -455,7 +455,7 @@ int acquire_terminal(
                                         goto fail;
                                 }
 
-                                r = fd_wait_for_event(fd, POLLIN, ts + timeout - n);
+                                r = fd_wait_for_event(notify, POLLIN, ts + timeout - n);
```

Comment:
It appears that the endless loop was caused by some debug traces I added. I do not really understand why since I made sure that the traces were emitted without closing the console...